### DISCUSSION TITLE
Swing

### DISCUSSION THREAD

**Discussion by 6809-Programmer** [hydrogen-music/hydrogen#1328] [Q&A]
The swing control is mentioned in Section 7.3 of the manual and resides in the Master section of the mixer. I assumed it was for creating the typical long, short-long, short-long cymbal swing pattern like what's used in jazz, but I couldn't get it to work, even after installing JazzFunkKit. There was no change in the relative length of eighth notes on the down-beats versus the up-beats. Of course, I could've setup a pattern based on triplets (3 notes per beat), but why if I could simply make it swing instead? Am I missing something here?

Hydrogen 1.0.2 on Windows 10.

**Comment by theGreatWhiteShark** [comment]
Hey @6809-Programmer ,

The *swing* does not alter the length of a note but places it slightly before the beat and is only affecting notes at predefined positions. When setting the resolution of a pattern to `1/16` every second note is affected. If set to `1/32` every third out of four consecutive notes will be affected. For smaller resolution the swing won't have any effect.

While you can control the amount of lead via the knob in the Mixer, please also notice that it depends on the current speed as well. The slower the song, the larger is the offset introduced.

**Reply by 6809-Programmer** [reply]
Oops, sorry. I did see that mentioned somewhere in the docs. So, it allows it to play slightly ahead or behind the beat. That's a nice effect and I agree that is useful if that is what you want. However, none of those resolutions, 1/8, 1/16, or 1/32, are subdivided into three parts. They are all based on dividing the beat in half, quarters, or eighths instead. I have come to the conclusion that creating a triplet swing feel (which is what I want) by subdividing the count (1, 2, 3, or 4) into three equal parts just isn't possible in Hydrogen because there are no resolutions such as 1/3, 1/6, 1/12. They can't divide the count into three equal parts. Alternatively as a work around, I tried changing the size to 12 or 24 and setting the resolution to 4, 8 or 16. That finally worked as it allowed me to group each count as 1=(1,2,3), count 2=(4,5,6), count 3=(7,8,9), count 4=(10,11,12), but it requires setting the tempo to THREE times the normal amount (360)! You can see a video of it in action here:

https://rumble.com/vk4d00-swinging-high-hat-rhythm-created-in-hydrogen-drum-machine-software.html

Thanks.

**Reply by cme** [reply]
Hydrogen does actuallly have triplet subdivisions, but thinking about it, it's probably hard to find that from the way it's presented in the GUI.

The "Res" drop-down menu has options "4T" "8T" etc, which set the grid resolution to quarter-note triplets, 8th-note triplets, etc.

Perhaps this would be more obvious if it actually said "triplets" rather than just "T"?

**Reply by theGreatWhiteShark** [reply]
> Perhaps this would be more obvious if it actually said "triplets" rather than just "T"?

In the upcoming version of Hydrogen (in `master`) "T" is already replaced by a translatable "triplet". But the word triplet is just used two times in the [current manual](http://hydrogen-music.org/documentation/manual/manual_en_single_page.html#chpt.pattern_editor). Maybe it should be more present?

**Reply by theGreatWhiteShark** [reply]
@6809-Programmer apart from the Swing button - which unarguably is not that flexible - you can also use the [Note Properties Ruler](http://hydrogen-music.org/documentation/manual/manual_en_single_page.html#chpt.pattern_editor.note_properties) to manually adjust the lead and lag of specific notes. This allows you to create a very custom and individual swing.

**Reply by 6809-Programmer** [reply]
> 
> 
> Hydrogen does actuallly have triplet subdivisions, but thinking about it, it's probably hard to find that from the way it's presented in the GUI.
> 
> The "Res" drop-down menu has options "4T" "8T" etc, which set the grid resolution to quarter-note triplets, 8th-note triplets, etc.
> 
> Perhaps this would be more obvious if it actually said "triplets" rather than just "T"?

Oh! I should have known that. As a newcomer and part of the learning curve to this software, I honestly did not know what the 'T' stood for. And the two different interpretations of the word 'Swing' only served to put a stumbling block in my learning curve. Hydrogen absolutely does have it the triplet subdivisions, and it's lot better than doing what I did. Thanks for letting me know.

**Comment by 6809-Programmer** [comment]
Solved!
